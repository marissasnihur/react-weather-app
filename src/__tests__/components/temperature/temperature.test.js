import React from 'react';
import ReactDOM from 'react-dom';

import Temperature from '../../../components/Temperature/Temperature';

describe('ForecastItem component', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Temperature>26</Temperature>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should render the prop value (temperature) correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Temperature>26</Temperature>, div);
    expect(div.firstChild.textContent).toBe('26');
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Temperature>26</Temperature>, div);
    expect(div).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
  });
  
});
