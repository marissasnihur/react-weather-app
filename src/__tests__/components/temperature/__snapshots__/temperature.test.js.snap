// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastItem component should match snapshot 1`] = `
<div>
  <div
    class="cardTemperature"
  >
    26
  </div>
</div>
`;
